packages/json-schema/src/decorators.ts

```diff
@@ -11,6 +11,7 @@ import {
   type Type,
```

Comment:
@btiernay thanks for trying to fix this, I think however this issue lies in `serializeValueToJson` and updat ethe json schema emitter to use that as well so it fixes it for everything. 

But as said on the issue we have to decide if that's something we want to do as it is.

Reply:
Got it, thanks!

Reply:
@timotheeguerin Thanks for getting back. I'm good with you folks taking this over as I'm sure the fix I've done here can be much improved 🙏 .